src/ProvidedTypes.fs

```diff
@@ -14044,25 +14044,59 @@ namespace ProviderImplementation.ProvidedTypes
         let stringTypeTgt = convTypeToTgt typeof<string>
         let mathTypeTgt = convTypeToTgt typeof<System.Math>
 
+#if FSHARP6

```

Comment:
I don't remember offhand, but will this be functionally equivalent to:

"if it's compiled using FSharp.Core 6.0"

or:

"if it's compiled using FSharp.Core 6.0 or greater"

If the former, then this probably needs to be a different condition. I know that .NET defines `_OR_GREATER` preprocessors, but I have no idea if F# does this. @vzarytovskii any thoughts?

Reply:
6 or greater: Just need support for Tasks. There would be multiple options: just conditions inside a file, or 2 separate files even and the condition on .fsproj level.

Looking the Nuget package stats of latest 6 weeks downloads, the F# core 5.0 and 4.x are still the most popular ones, and there are TPs on F# 4 getting thousands of downloads a month. So we definitely not want to break F# 4 support.